src/payment/strategies/bolt/bolt-app-payment-strategy.spec.ts

```diff
@@ -0,0 +1,187 @@
+  let submitPaymentAction: Observable<SubmitPaymentAction>;
+  let payload: OrderRequestBody;
+  let paymentActionCreator: PaymentActionCreator;
+  let paymentHumanVerificationHandler: PaymentHumanVerificationHandler;
+  let paymentMethodActionCreator: PaymentMethodActionCreator;
+  let paymentMethodMock: PaymentMethod;
+  let paymentMethodRequestSender: PaymentMethodRequestSender;
+  let paymentRequestSender: PaymentRequestSender;
+  let paymentRequestTransformer: PaymentRequestTransformer;
+
+  beforeEach(() => {
+      store = createCheckoutStore(getCheckoutStoreState());
+      paymentMethodMock = getBolt();
+      const scriptLoader = createScriptLoader();
+      requestSender = createRequestSender();
+      orderActionCreator = new OrderActionCreator(
+          new OrderRequestSender(requestSender),
+          new CheckoutValidator(new CheckoutRequestSender(requestSender))
+      );
+
+      paymentRequestTransformer = new PaymentRequestTransformer();
+      paymentRequestSender = new PaymentRequestSender(createPaymentClient());
+      paymentHumanVerificationHandler = new PaymentHumanVerificationHandler(createSpamProtection(createScriptLoader()));
+      paymentActionCreator = new PaymentActionCreator(
+          paymentRequestSender,
+          orderActionCreator,
+          paymentRequestTransformer,
+          paymentHumanVerificationHandler
+      );
+      submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+      submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+      paymentMethodRequestSender = new PaymentMethodRequestSender(requestSender);
+      paymentMethodActionCreator = new PaymentMethodActionCreator(paymentMethodRequestSender);
+      boltClient = getBoltScriptMock(true);
+      boltScriptLoader = new BoltScriptLoader(scriptLoader);
+
+      payload = {
+          payment: {
+              methodId: 'bolt',
+              paymentData: {
+                  nonce: 'fooNonce',
+              },
+          },
+      };
+      options = { methodId: 'bolt' };
+
+      jest.spyOn(store, 'dispatch');
+
+      jest.spyOn(orderActionCreator, 'submitOrder')
+          .mockReturnValue(submitOrderAction);
+      jest.spyOn(paymentActionCreator, 'submitPayment')
+          .mockReturnValue(submitPaymentAction);
+      jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+          .mockResolvedValue(store.getState());
+      jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+          .mockReturnValue(paymentMethodMock);
+      jest.spyOn(boltScriptLoader, 'load')
+          .mockResolvedValue(boltClient);
+
+      strategy = new BoltAppPaymentStrategy(
+        store,
+        orderActionCreator,
+        paymentActionCreator,
+        paymentMethodActionCreator,
+        boltScriptLoader
+      );
+  });
+
+  describe('#initialize', () => {
+    it('successfully initializes the bolt strategy and loads the bolt client', async () => {
```

Comment:
this file seems to have a mix of indentation styles, we use 4 spaces for indentation.